### PR TITLE
Update frontend tests to remove the errors

### PR DESCRIPTION
**Pull request by helenatran**
This PR only update the tests to remove the errors from the changes of codes we have made in the TS files. This PR does not try to make the tests pass, but just to get rid of all the errors which were in the frontend. 

Note: It seems that the 'google is undefined' error is back 😢  I think it might be because we call the mapInitialiser in the ngAfterViewInit(), but I am not sure at all.

### CODE REVIEW

File: src/main/webapp/src/app/hidden-gem.service.spec.ts [hunk 1]
```diff
@@ -110,35 +96,28 @@ describe('HiddenGemService', () => {
     const expectedTopGems: HiddenGem[] =
```
Comment (sehhajdkg) [inline src/main/webapp/src/app/hidden-gem.service.spec.ts:96]: Maybe we don't need this test anymore. We initially had a 2 tests for the http call for ranking and another for recommendations. Now that its all handled in the one hidden gems call, I feel like this test might be redundant. It's basically testing the same thing as the first one?

File: src/main/webapp/src/app/hidden-gem.service.spec.ts [hunk 2]
```diff
@@ -153,7 +132,7 @@ describe('HiddenGemService', () => {
 
     httpClientSpy.get.and.returnValue(of(errorResponse));
 
-    hiddenGemService.findHiddenGemRecommendation("").subscribe(
+    hiddenGemService.getRankedHiddenGems(LOCATION).subscribe(
```
Comment (sehhajdkg) [inline src/main/webapp/src/app/hidden-gem.service.spec.ts:135]: this error test initially for recommendation can also be removed i think. Its a duplicate test

File: src/main/webapp/src/app/hidden-gem.service.spec.ts (hunk 1 above)
Comment (helenatran) [reply src/main/webapp/src/app/hidden-gem.service.spec.ts]: That's a good point! I removed the 2 duplicate tests :)

File: src/main/webapp/src/app/hidden-gem.service.spec.ts [hunk 3]
```diff
@@ -104,62 +90,4 @@ describe('HiddenGemService', () => {
     );
     expect(httpClientSpy.get.calls.count()).toBe(1, 'one call');
   })
-
-  it('should return hidden gem recommendations', () => {
```
Comment (jamila-sabaz) [inline src/main/webapp/src/app/hidden-gem.service.spec.ts:108]: this might be a silly question but why are we removing this test?

**Review by jamila-sabaz — APPROVED** [review]

**Review by jamila-sabaz — COMMENTED** [review]
looks good, we can merge it asap so Sehhaj and I can continue working on tests without errors due to changed interface

**Review by lorikeet-s — APPROVED** [review]

**Pull request merged**